docs/defs.md

```diff
@@ -0,0 +1,72 @@
+<!-- Generated with Stardoc: http://skydoc.bazel.build -->
+
+Starlark rules for building protocol buffers.
+
+<a id="proto_descriptor_set"></a>
+
+## proto_descriptor_set
+
+<pre>
+proto_descriptor_set(<a href="#proto_descriptor_set-name">name</a>, <a href="#proto_descriptor_set-deps">deps</a>)
+</pre>
+
+Collects all `FileDescriptorSet`s from `deps` and combines them into a single
+`FileDescriptorSet` containing all the `FileDescriptorProto`.
+
+**ATTRIBUTES**
+
+
+| Name  | Description | Type | Mandatory | Default |
+| :------------- | :------------- | :------------- | :------------- | :------------- |
+| <a id="proto_descriptor_set-name"></a>name |  A unique name for this target.   | <a href="https://bazel.build/concepts/labels#target-names">Name</a> | required |  |
+| <a id="proto_descriptor_set-deps"></a>deps |  Sequence of <code>ProtoInfo</code>s to collect <code>FileDescriptorSet</code>s from.   | <a href="https://bazel.build/concepts/labels">List of labels</a> | optional | <code>[]</code> |
+
+
+<a id="proto_lang_toolchain"></a>
+
+## proto_lang_toolchain
+
+<pre>
+proto_lang_toolchain(<a href="#proto_lang_toolchain-name">name</a>, <a href="#proto_lang_toolchain-toolchain_type">toolchain_type</a>, <a href="#proto_lang_toolchain-exec_compatible_with">exec_compatible_with</a>, <a href="#proto_lang_toolchain-target_compatible_with">target_compatible_with</a>, <a href="#proto_lang_toolchain-attrs">attrs</a>)
+</pre>
+
+Creates a proto_lang_toolchain and corresponding toolchain target.
+
+Toolchain target is only created when toolchain_type is set.
+
+https://docs.bazel.build/versions/master/be/protocol-buffer.html#proto_lang_toolchain
+
+
+**PARAMETERS**
+
+
+| Name  | Description | Default Value |
+| :------------- | :------------- | :------------- |
+| <a id="proto_lang_toolchain-name"></a>name |  name of the toolchain   |  none |
+| <a id="proto_lang_toolchain-toolchain_type"></a>toolchain_type |  The toolchain type   |  <code>None</code> |
+| <a id="proto_lang_toolchain-exec_compatible_with"></a>exec_compatible_with |  ([constraints]) List of constraints the prebuild binaries is compatible with.   |  <code>[]</code> |
+| <a id="proto_lang_toolchain-target_compatible_with"></a>target_compatible_with |  ([constraints]) List of constraints the target libraries are compatible with.   |  <code>[]</code> |
+| <a id="proto_lang_toolchain-attrs"></a>attrs |  Rule attributes   |  none |
+
+
+<a id="proto_library"></a>
+
+## proto_library
+
+<pre>
+proto_library(<a href="#proto_library-attrs">attrs</a>)
+</pre>
+
+Bazel proto_library rule.
+
+https://docs.bazel.build/versions/master/be/protocol-buffer.html#proto_library
+
+
+**PARAMETERS**
+
+
+| Name  | Description | Default Value |
+| :------------- | :------------- | :------------- |
+| <a id="proto_library-attrs"></a>attrs |  Rule attributes   |  none |
```

Comment:
Shouldn't there be `srcs`, `deps` attribute description in `proto_library`?

Reply:
I don't think stardoc can generate correct documentation for native rules. this should be fixed once there's a starlark proto_library

Reply:
Ah I see, it's actually a wrapper macro. You could add attrs and documentation to it, but the documentation of a macro is also different from documentation of a rule. 

I see this PR as an improvement - because at least we are generating the docs. But not a big one for the users, because the generated docs are useless. 

To address the users needs better, it makes sense to copy the docs from BazelProtoLibrary.java into the macro and add srcs, deps attribute. Completely ok to do this in a separate PR.

To answer your question, there's a native `starlark_doc_extract` rule, that can extract documentation from proto_library.bzl. But you'd need to directly refer to proto_library rule (without macro). You could also have a helper.bzl file that does that, for the purpose of documentation only (so that you can keep the macro).  But eventually you'd find out that builtin proto_library.bzl is not documented. That's because the docs on build encyclopedia are still generated from `BazelProtoLibrary.java`